pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/inv/geospatial/BaseH3IndexCreator.java

```diff
@@ -101,6 +102,11 @@ public abstract class BaseH3IndexCreator implements GeoSpatialIndexCreator {
     _lowestResolution = resolution.getLowestResolution();
   }
 
+  @Override
+  public Geometry deserialize(byte[] bytes) {
```

Comment:
This is another case where I decided to do not touch the code we already have. There are several usages of each constant in V1Constants around the code. We can refactor them to use the value returned by each `IndexType.getFileExtension` but it would require to provide the `ColumnMetadata` (at least if we use the general method).

As said, I preferred to do not modify the code we already have if there is no need to do that in order to reduce the number of changes. I would vote to apply the change, but in new PRs.

Reply:
I see. I was looking at IndexType.Dict.INSTANCE.getFileExtension() and didn't notice there is a more general one getFileExtension(ColumnMetadata).

Reply:
It isn't great. All indexes but forward always return a constant. But forward use different extensions depending on whether the column is multivalued, raw, sorted, etc.